### PR TITLE
Small Changes to bounded sharing

### PR DESCRIPTION
**Pull request by calebmkim**
Gives more flexibility to the bound settings. 

(For example now we can not share combinational cells, share registers 5 times, and always share other components).

### CODE REVIEW

File: tests/passes/cell-share/bounded.futil [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// -p cell-share -x cell-share:bounds=2,3,4 -p remove-ids
```
Comment (rachitnigam) [inline tests/passes/cell-share/bounded.futil:1]: What does it mean to have a negative bound?

Comment (calebmkim) [reply tests/passes/cell-share/bounded.futil:1]: it means always share that cell type.

Comment (rachitnigam) [reply tests/passes/cell-share/bounded.futil:1]: Got it. If it's not already documented, we should say that in the comment for the pass.

**Comment by rachitnigam** [comment]
LGTM!

**Pull request merged**